clients/rust/tests/inactivate_stake.rs

```diff
@@ -96,11 +98,213 @@ async fn inactivate_stake() {
     assert_eq!(stake_account.amount, 50);
     assert_eq!(stake_account.deactivating_amount, 0);
     assert_eq!(stake_account.inactive_amount, 50);
+    assert!(stake_account.deactivation_timestamp.value().is_none());
 
-    // And the total delegated must decrease.
+    // And the total delegated on the config must decrease.
 
     let account = get_account!(context, config);
     let config_account = Config::from_bytes(account.data.as_ref()).unwrap();
 
     assert_eq!(config_account.token_amount_delegated, 50);
 }
+
+#[tokio::test]
+async fn inactivate_stake_with_no_deactivated_amount() {
+    let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+        .start_with_context()
+        .await;
+
+    // Given a config account (total amount delegated = 100).
+
+    let config = create_config(&mut context).await;
+    // "manually" set the total amount delegated
+    let account = get_account!(context, config);
+    let mut config_account = Config::from_bytes(account.data.as_ref()).unwrap();
+    config_account.token_amount_delegated = 100;
+
+    let updated_config = Account {
+        lamports: account.lamports,
+        data: config_account.try_to_vec().unwrap(),
+        owner: account.owner,
+        executable: account.executable,
+        rent_epoch: account.rent_epoch,
+    };
+    context.set_account(&config, &updated_config.into());
+
+    // And a stake account (amount = 100).
+
+    let validator = Pubkey::new_unique();
+    let authority = Keypair::new();
+    let vote = create_vote_account(&mut context, &validator, &authority.pubkey()).await;
+
+    let stake_pda = create_stake(&mut context, &validator, &vote, &config).await;
+
+    let account = get_account!(context, stake_pda);
+    let mut stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the stake values
+    stake_account.amount = 100;
+
+    let updated_stake = Account {
+        lamports: account.lamports,
+        data: stake_account.try_to_vec().unwrap(),
+        owner: account.owner,
+        executable: account.executable,
+        rent_epoch: account.rent_epoch,
+    };
+    context.set_account(&stake_pda, &updated_stake.into());
+
+    // When we try to inactivate the stake without any deactivated amount.
+
+    let inactivate_ix = InactivateStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[inactivate_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the tranasction succeeds and the stake account should
+    // remain the same.
```

Comment:
As per my other comments on the processor, I definitely feel it makes sense to have an error here.